### PR TITLE
Add CORS headers to the Health handler

### LINKED ISSUE #13935: Allow cross origin requests on health endpoints
**Description**
Cross origin requests on a health endpoint such as `/health/ready` fail as the response does not contain the required CORS headers.

This is the error message from console in chrome: `Access to XMLHttpRequest at 'https://...' from origin 'https://...' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.`

This was supported in thorntail. See https://github.com/thorntail/thorntail/pull/745

**Implementation ideas**
Add required CORS headers by default or introduce a new configuration property which, when true, allows cross origin requests on health endpoints.

### PR DESCRIPTION
**Pull request by jmartisk**
fixes https://github.com/quarkusio/quarkus/issues/13935

### CODE REVIEW

File: extensions/smallrye-health/runtime/src/main/java/io/quarkus/smallrye/health/runtime/SmallRyeHealthHandlerBase.java [hunk 1]
```diff
@@ -18,6 +20,16 @@
 
     protected abstract SmallRyeHealth getHealth(SmallRyeHealthReporter reporter, RoutingContext routingContext);
 
+    private static final Map<String, String> corsHeaders;
+
+    static {
+        corsHeaders = new HashMap<>();
+        corsHeaders.put("Access-Control-Allow-Origin", "*");
```
Comment (gsmet) [inline extensions/smallrye-health/runtime/src/main/java/io/quarkus/smallrye/health/runtime/SmallRyeHealthHandlerBase.java:27]: Do we really want to allow everything and not make it configurable?

That seems a bit dangerous even if health information is probably not that critical. Maybe it's not worth it, I'll let you judge of that :).

Comment (sberyozkin) [reply extensions/smallrye-health/runtime/src/main/java/io/quarkus/smallrye/health/runtime/SmallRyeHealthHandlerBase.java:27]: Yeah - we already have a CORS filter which should be usable ?

**Comment by jmartisk** [comment]
Oh, I wasn't aware that we have a general CORS filter that can be used for this. Thanks @sberyozkin for letting me know. In that case, this PR is probably not necessary.

**Comment by sberyozkin** [comment]
Hi @jmartisk Np at all, give it a try please, https://quarkus.io/guides/http-reference#cors-filter

I suppose what we'd like to have is a single set of properties affecting the CORS polict across all the deployment. With those properties you can also make it all more secure, indeed, as Guillaume pointed out, a wildcard is generally problematic, and yeah, I know I used to do it myself a few times as well :-)

**Pull request closed without merging**